backend/src/main/java/elbin_bank/issue_tracker/issue/infrastructure/query/strategy/AssigneeFilterStrategy.java

```diff
@@ -0,0 +1,32 @@
+package elbin_bank.issue_tracker.issue.infrastructure.query.strategy;
+
+import elbin_bank.issue_tracker.issue.application.query.dsl.FilterCriteria;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AssigneeFilterStrategy implements FilterStrategy {
+
+    @Override
+    public boolean supports(FilterCriteria c) {
+        return !c.assignees().isEmpty();
+    }
+
+    @Override
+    public void applyJoin(StringBuilder join, FilterCriteria c) {
+        join.append(" JOIN assignee ass ON ass.issue_id = i.id")
+                .append(" JOIN `user` u2 ON u2.id = ass.user_id");
+    }
```

Comment:
```suggestion
    @Override
    public StringBuilder applyJoin(StringBuilder join, FilterCriteria c) {
        return join.append(" JOIN assignee ass ON ass.issue_id = i.id")
                .append(" JOIN `user` u2 ON u2.id = ass.user_id");
    }
```
현재 `applyJoin` 메서드는 `StringBuilder`에 직접 side effect를 일으키는 방식인데요, 개인적으로는 반환값을 명시해서 사용하는 패턴을 더 선호합니다.
이렇게 되면 함수 시그니처만 보더라도 해당 함수가 StringBuilder를 수정하고 반환한다는 동작이 드러나기 때문에, 의도를 파악하기 더 쉬워지고 테스트 측면에서도 유용합니다.

현재처럼 void 반환을 사용하면 내부 상태를 알기 어려워지고, 체이닝이 불가능해져서 빌더 패턴처럼 유연하게 쓸 수 없다는 단점도 있습니다.

Reply:
오 참고해보겠습니다 !